### ISSUE TITLE
Group buttons produce no data in version `14.15.3`

### ISSUE THREAD

**Issue by forgetfulskybro** [discordjs/discord.js#10431]
### Which package is this bug report for?

discord.js

### Issue description

When updating to version `14.15.3` of Discord.js I've encountered an issue when running button interactions in Group DMs. Buttons return no data to `interactionCreate` when logging all interactions. Slashcommands work as expected. As soon as I downgraded to version `14.15.2` button interactions worked perfectly. 

This can be reproduced by running the latest version of Discord.js and using a button inside of a Group DM. 

For possible issues on our part, here's the source code: [Would You](https://github.com/Would-You-Bot/client/blob/main/src/events/interactionCreate.ts)

### Code sample

```typescript
import { ButtonInteraction, Interaction } from "discord.js";
import buttonInteractionEvent from "../buttons";
import commandInteractionEvent from "../commands";
import { Event } from "../interfaces";
import WouldYou from "../util/wouldYou";

const event: Event = {
  event: "interactionCreate",
  execute: async (
    client: WouldYou,
    interaction: Interaction | ButtonInteraction,
  ) => {
    console.log(interaction) // Button produced no data and nothing was logged
    if (interaction.isCommand()) {
      await commandInteractionEvent.execute(client, interaction);
    } else if (interaction.isButton() || interaction.isAnySelectMenu()) {
      await buttonInteractionEvent.execute(client, interaction);
    }
  },
};

export default event;
```

### Versions

NodeJS version: v22.2.0
discord.js: 14.15.3
@discordjs/rest: 2.3.0
discord-api-types: 0.37.91
Windows version: Windows 11 Home Insider Preview 23H2, Build 26040.1000

### Issue priority

Low (slightly annoying)

### Which partials do you have configured?

Channel

### Which gateway intents are you subscribing to?

Guilds, GuildMembers, GuildMessages

### I have tested this issue on a development release

Not tested

**Comment by forgetfulskybro** [comment]
Adding @mezotv to issue

**Comment by Qjuh** [comment]
This is a known issue and will be fixed by #10348

**Comment by forgetfulskybro** [comment]
> This is a known issue and will be fixed by #10227

Alright, thanks for showing me this PR
